### ISSUE TITLE
Post processor @Bean should be marked static

### ISSUE THREAD

**Issue by charlesrgould** [ulisesbocchio/jasypt-spring-boot#16]
Thanks for creating this project. Quick observation.

Starting an application gives me this warning:

```
WARN org.springframework.context.annotation.ConfigurationClassEnhancer - 
  @Bean method EnableEncryptablePropertySourcesConfiguration
    .enableEncryptablePropertySourcesPostProcessor is non-static and returns
    an object assignable to Spring's BeanFactoryPostProcessor interface. This
    will result in a failure to process annotations such as @Autowired,
    @Resource and @PostConstruct within the method's declaring
    @Configuration class. Add the 'static' modifier to this method to avoid these
    container lifecycle issues; see @Bean javadoc for complete details
```

See [@Bean Javadoc](http://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/context/annotation/Bean.html), section titled `BeanFactoryPostProcessor-returning @Bean methods`

**Comment by charlesrgould** [comment]
[https://github.com/ulisesbocchio/jasypt-spring-boot/blob/master/jasypt-spring-boot/src/main/java/com/ulisesbocchio/jasyptspringboot/configuration/EnableEncryptablePropertySourcesConfiguration.java#L66](https://github.com/ulisesbocchio/jasypt-spring-boot/blob/master/jasypt-spring-boot/src/main/java/com/ulisesbocchio/jasyptspringboot/configuration/EnableEncryptablePropertySourcesConfiguration.java#L66)

I see that the post processor uses the (non-static) environment variable.

I suppose since there are no other annotated methods in the class to worry about, it should be fine. Would be nice to avoid that warning though. If not, maybe a comment in the class explaining this pitfall?

**Comment by ulisesbocchio** [comment]
Yeah, I'm aware of this. It's a Spring Lifecycle issue that doesn't affect the lifecycle of the beans of this project. You can shut off the warning by adding this to your logback.xml:
```
<logger name="org.springframework.context.annotation.ConfigurationClassEnhancer" level="ERROR"/>
```
I'm looking into a different way to bootstrap Jasypt intro Spring Boot that will eventually avoid this issue.

**Comment by ulisesbocchio** [comment]
This is fix as of version 1.6, enjoy

**Comment by charlesrgould** [comment]
Thanks!
